### ISSUE TITLE
Please allow a list instead of a tuple in new intersphinx_mapping format too!

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#978]
The documentation at http://sphinx.pocoo.org/ext/intersphinx.html#confval-intersphinx_mapping describes the new intersphinx_mapping format. It uses tuples as in this example:

```
#!python

intersphinx_mapping = {'python': ('http://docs.python.org/3.2', None)}
```

I kindly ask to allow a list as well instead of a tuple only. Currently a list leads to the exception: "<type 'exceptions.AttributeError'>: 'list' object has no attribute 'find'".

The reason is that we (TYPO3 community) want to use  Settings.yml file in YAML format to. I'd like to use the yaml.safe_load(...) method of PyYAML. There I can easily load lists but not tuples.

What about going for the rule in general in conf.py which says: "You may use lists instead of tuples too."?

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/978
- Originally reported by: Martin Bless
- Originally created at: 2012-07-25T15:42:26.952

**Comment by shimizukawa** [comment]
_From Martin Bless on 2012-07-25 19:05:36+00:00_

It looks like (untested!) that the only thing that needs to be changed is:
In "sphinx.ext.intersphinx.py", function "def load_mappings(app):"

old line:
  if isinstance(value, tuple):

new line:
  if isinstance(value, (tuple, list)):

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-10-30 03:56:29+00:00_

I think so too and this change have no side-effect.

location is here: [ext/intersphinx.py L-164](https://github.com/sphinx-doc/testing/blob/0c02841412f8c0efe871853f192252e58e23680d/sphinx/ext/intersphinx.py#L164)

**Comment by tk0miya** [comment]
Fixed at 	175c6e6.
